### PR TITLE
Remove the -p flag for pilot test.

### PR DESCRIPTION
**Pull request by jmuk**
I found that https://github.com/istio/istio/issues/2318 was the
reason why, but actually -p flag doesn't deal with the reported
problem.

The -p flag specifies the number of processes `go test` runs
simultaneously, while the test flakiness seems happening purely
within a single process, and no other running processes seems not
to interfere the reported test case. The error would be affected
by the goroutine scheduling, but this flag is unrelated.

On the other hand, removing this flag accelerate the test time
since `go test` can run more tests at the same time.

### CODE REVIEW

**Comment by costinm** [comment]
The -p is needed - in low resource vm compile will fail, too many processes
compiling. Please keep it.

**Comment by jmuk** [comment]
If so, can we make it as a part of the test infra config rather than having the default value in Makefile?

Also, why it's pilot only then? That sounds universal to me.

**Comment by kyessenov** [comment]
use cpu_num?
or at least override in circle ci only?

**Comment by jmuk** [comment]
Is it circleci only thing?

**Comment by jmuk** [comment]
added circle-ci config

**Comment by costinm** [comment]
It is not circleci specific (and in fact it doesn't happen in our circle ci - since we get xlarge).

Any machine with lower RAM - or the 'free circleci' ( for people who fork istio ) would be affected. I think there is a bug filled for go - the problem it's that by default it starts too many compiles.

Only for pilot because for some reasons it only failed pilot test..

Ok to make it optional and document to use in case of errors or low mem .

File: .circleci/config.yml [hunk 1]
```diff
@@ -272,7 +272,7 @@ jobs:
             free
             trap "go-junit-report </go/out/tests/go-test-report.out > /go/out/tests/go-test-report.xml" EXIT
             make localTestEnv
-            make pilot-test mixer-test broker-test security-test T=-v | tee -a /go/out/tests/go-test-report.out
+            make pilot-test mixer-test broker-test security-test common-test T="-v -p 1" | tee -a /go/out/tests/go-test-report.out
```
Comment (rshriram) [inline .circleci/config.yml]: This will make it pointless to have a large instance and run only one test at a time

Comment (costinm) [reply .circleci/config.yml]: The proper way is to use the CI 'context' env variables - we can leave the default matching a large image.

The problem is that even on a large image I've seen failures - I don't think the default (guessed by go) is good enough. But we can adjust.

**Comment by kyessenov** [comment]
Can we please remove -p restriction on non-circle ci build?
I have 32 core 64 gig machine, and it's unnecessary to slow it down.

**Comment by kyessenov** [comment]
Can we please not require a bunch of env variables, just to get a normal
build working?
How about we do the opposite and make circleci set CIRCLE_CI so that
makefile can do its quirks.

**Comment by costinm** [comment]
This has nothing to do with circle ci - but with getting the tests to pass
on VMs with less RAM.
There is a bug in golang, where you 'go test' large packages it seems to
spawn more processes than
the machine can handle.

Nothing to do with the official circleci either - which is also x-large in
RAM, but not everyone
who builds will have this.

**Pull request closed without merging**